ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -376,28 +376,37 @@ private[kvutils] class TransactionCommitter(
   private def buildFinalLogEntry: Step =
     (commitContext, transactionEntry) => StepStop(buildLogEntry(transactionEntry, commitContext))
 
-  private def validateContractKeys: Step = (commitContext, transactionEntry) => {
+  private[committer] def validateContractKeys: Step = (commitContext, transactionEntry) => {
     val damlState = commitContext.collectInputs {
       case (key, Some(value)) if key.hasContractKey => key -> value
     } ++ commitContext.getOutputs
     val startingKeys = damlState.collect {
       case (k, v) if k.hasContractKey && v.getContractKeyState.getContractId.nonEmpty => k
     }.toSet
-    validateContractKeyUniqueness(commitContext.recordTime, transactionEntry, startingKeys) match {
-      case StepContinue(transactionEntry) =>
-        validateContractKeyCausalMonotonicity(
-          commitContext.recordTime,
-          transactionEntry,
-          startingKeys,
-          damlState)
-      case err => err
+    val resolvedIds: Map[DamlContractKey, String] = damlState.collect {
+      case (k, v) if k.hasContractKey && v.getContractKeyState.getContractId.nonEmpty =>
+        k.getContractKey -> v.getContractKeyState.getContractId
+    }
+
+    List(
+      validateContractKeyUniqueness(startingKeys)(_, _),
+      validateContractKeyCausalMonotonicity(startingKeys, damlState)(_, _),
+      validateContractKeysConsistency(resolvedIds)(_, _),
```

Comment:
None of these methods modify `transactionEntry`. So I wonder why you pass the `transactionEntry` around as the state of the fold. You could use `()` instead and directly pass `transactionEntry` to the methods.

Also, I'm not sure why you need to explicityly eta-expand these methods with `(_, _)`.

Reply:
Good point! I'll modify this.

Reply:
Fixed.